adafruit_display_shapes/triangle.py

```diff
@@ -0,0 +1,218 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2019 Limor Fried for Adafruit Industries
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`triangle`
+================================================================================
+
+Various common shapes for use with displayio - Triangle shape!
+
+
+* Author(s): Melissa LeBlanc-Williams
+
+Implementation Notes
+--------------------
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://github.com/adafruit/circuitpython/releases
+
+"""
+
+import displayio
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_Display_Shapes.git"
+
+
+class Triangle(displayio.TileGrid):
+    # pylint: disable=too-many-arguments,invalid-name
+    """A triangle.
+
+    :param x0: The x-position of the first vertex.
+    :param y0: The y-position of the first vertex.
+    :param x1: The x-position of the second vertex.
+    :param y1: The y-position of the second vertex.
+    :param x2: The x-position of the third vertex.
+    :param y2: The y-position of the third vertex.
+    :param fill: The color to fill the triangle. Can be a hex value for a color or
+                 ``None`` for transparent.
+    :param outline: The outline of the triangle. Can be a hex value for a color or
+                    ``None`` for no outline.
+    """
+    def __init__(self, x0, y0, x1, y1, x2, y2, *, fill=None, outline=None):
+        # Sort coordinates by Y order (y2 >= y1 >= y0)
+        if y0 > y1:
+            y0, y1 = y1, y0
+            x0, x1 = x1, x0
+
+        if y1 > y2:
+            y1, y2 = y2, y1
+            x1, x2 = x2, x1
+
+        if y0 > y1:
+            y0, y1 = y1, y0
+            x0, x1 = x1, x0
+
+        smallest_x = x0
+        largest_x = x0
+
+        # Find the largest and smallest X values to figure out width for bitmap
```

Comment:
Please use min and max to simplify this logic:

```python
xs = [x0, x1, x2]
width = max(xs) - min(xs) + 1
```

Reply:
Nice! I just learned something :)